ui/components/app/currency-input/currency-input.js

```diff
@@ -112,14 +111,8 @@ export default function CurrencyInput({
     );
   };
 
-  // reset form when token is changed
+  // align input to upstream value
   useEffect(() => {
-    setTokenDecimalValue('0');
-    setFiatDecimalValue('0');
-  }, [asset?.address]);
-
```

Comment:
I don't think we can fully remove this bc it reintroduces this bug
1. Enter 1 ETH
2. Toggle to fiat
3. Change token to USDC
4. Input amount is not updated

https://github.com/MetaMask/metamask-extension/assets/100321200/1256e4bf-4cdf-455f-81d1-44031213165d

Reply:
addressed this in my latest commit